src/saltext/github/states/github.py

```diff
@@ -743,3 +743,175 @@ def repo_absent(name, profile="github", **kwargs):
             ] = f"Failed to delete repo {name}. Ensure the delete_repo scope is enabled if using OAuth."
             ret["result"] = False
     return ret
+
+
+def ruleset_absent(
+    name, ruleset_type, profile="github", owner=None, repo_name=None, org_name=None, **kwargs
+):
+    """
+    Ensure a ruleset is absent.
+
+    Example:
+
+    .. code-block:: yaml
+
+        ensure rulset test is absent in github:
+            github.ruleset_absent:
+                - name: 'test'
+                - ruleset_type: 'repo'
+                - owner: test
+                - repo_name: test_repo
+
+    The following parameters are required:
+
+    name
+        name of the ruleset.
+    ruleset_type
+        type of ruleset ('org' or 'repo')
+    """
+
+    ret = {"name": name, "changes": {}, "result": None, "comment": ""}
+
+    if ruleset_type not in ["org", "repo"]:
+        raise CommandExecutionError
+
+    if ruleset_type == "repo":
+        kwargs.update({"owner": owner, "repo_name": repo_name, "ruleset_type": ruleset_type})
+    if ruleset_type == "org":
+        kwargs.update({"org_name": org_name, "ruleset_type": ruleset_type})
+
+    try:
+        rulesets = __salt__["github.list_rulesets"](profile, **kwargs)
+    except CommandExecutionError:
+        rulesets = None
+
+    if rulesets:
+        for ruleset in rulesets:
+            if ruleset["name"] == name:
+                if __opts__["test"]:
+                    ret["comment"] = f"Ruleset {name} will be deleted"
+                    ret["result"] = None
+                    return ret
+
+                kwargs.update({"ruleset_id": ruleset["id"]})
+                result = __salt__["github.delete_ruleset"](profile, **kwargs)
+
+                if result:
+                    ret["comment"] = f"Deleted ruleset {name}"
+                    ret["changes"].setdefault("old", f"ruleset {name} exists")
+                    ret["changes"].setdefault("new", f"ruleset {name} deleted")
```

Comment:
**question (non-blocking):** why are we using `.setdefault()` here? it doesn't look like the `changes` key is set anywhere else, so shouldn't we be able to simplify?

```suggestion
                    ret["changes"]["old"] = f"ruleset {name} exists")
                    ret["changes"]["new"] = f"ruleset {name} deleted")
```